src/TableView.jl

```diff
@@ -19,12 +19,27 @@ end
 to_css_size(s::AbstractString) = s
 to_css_size(s::Real) = "$(s)px"
 
-function showtable(table; dark = false, height = :auto, width = "100%")
-    if !Tables.istable(typeof(table))
-        throw(ArgumentError("Argument is not a table."))
+struct IteratorAndFirst{F, T}
+    first::F
+    source::T
+    function IteratorAndFirst(x)
+        first = iterate(x)
+        return new{typeof(first), typeof(x)}(first, x)
     end
+end
+Base.IteratorSize(::Type{IteratorAndFirst{F, T}}) where {F, T} = Base.IteratorSize(T)
+Base.length(x::IteratorAndFirst) = length(x.source)
```

Comment:
Is the source guaranteed to keep it's length constant even if you iterate the first element away?

Reply:
Ah, good catch! Let me correct this